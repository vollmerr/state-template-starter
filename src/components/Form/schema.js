import { fieldOptions } from 'state-template';

const section1 = {
  exampleInput: {
    required: true,
    name: 'exampleInput',
    label: 'Example Input',
  },
  exampleDate: {
    name: 'exampleDate',
    label: 'Example Date',
  },
  exampleSelect: {
    name: 'exampleSelect',
    label: 'Example Select',
    options: fieldOptions.countyOptionsCA,
  },
  exampleCombobox: {
    name: 'exampleCombobox',
    label: 'Example ComboBox',
    options: fieldOptions.countyOptionsCA,
  },
};

const section2 = {
  exampleRadios: {
    name: 'exampleRadios',
    label: 'Example Radios',
    options: fieldOptions.yesNoOptions,
  },
  exampleCheckboxes: {
    name: 'exampleCheckboxes',
    label: 'Example Checkboxes ',
    options: fieldOptions.yesNoOptions,
  },
};

const schema = {
  section1,
  section2,
};

export default schema;
