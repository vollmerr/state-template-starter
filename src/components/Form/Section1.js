import React from 'react';
import { FieldInput, FieldSelect, FieldComboBox } from 'state-template';

import schema from './schema';

const { section1 } = schema;

const Section1 = () => (
  <div className={'row'}>
    <h2 className={'col-12'}>Section 1</h2>
    <FieldInput {...section1.exampleInput} className={'col-md-6'} />
    <FieldInput {...section1.exampleDate} className={'col-md-6'} />
    <FieldSelect {...section1.exampleSelect} className={'col-md-6'} />
    <FieldComboBox {...section1.exampleCombobox} className={'col-md-6'} />
  </div>
);

export default Section1;
