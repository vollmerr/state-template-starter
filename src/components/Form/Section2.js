import React from 'react';
import { FieldRadioButtons, FieldCheckboxes } from 'state-template';

import schema from './schema';

const { section2 } = schema;

const Section2 = () => (
  <div className={'row'}>
    <h2 className={'col-12'}>Section 2</h2>
    <FieldRadioButtons {...section2.exampleRadios} className={'col-md-6'} />
    <FieldCheckboxes {...section2.exampleCheckboxes} className={'col-md-6'} />
  </div>
);

export default Section2;
