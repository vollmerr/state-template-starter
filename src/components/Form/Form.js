import React from 'react';
import { Page, Form as ReduxForm, Button } from 'state-template';

import Section1 from './Section1';
import Section2 from './Section2';

export const Form = () => (
  <Page title={'Form'}>
    <ReduxForm
      form={'exampleName'}
      onSubmit={(values) => { alert(JSON.stringify(values, null, 2)); }}
    >
      <Section1 />
      <Section2 />

      <Button
        type={'submit'}
        text={'Submit'}
        variant={'primary'}
      />
    </ReduxForm>
  </Page>
);

export default Form;
