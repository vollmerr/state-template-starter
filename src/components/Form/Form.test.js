import React from 'react';
import { shallow } from 'enzyme';

import Form from './Form';

const props = {};

let wrapper;
describe('Form', () => {
  beforeEach(() => {
    wrapper = shallow(<Form {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
