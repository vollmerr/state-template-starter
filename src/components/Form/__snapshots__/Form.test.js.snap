// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<Page
  bannerSrc={null}
  className={null}
  title="Form"
>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="exampleName"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    onSubmit={[Function]}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  >
    <Section1 />
    <Section2 />
    <Button
      active={false}
      baseClass="btn"
      className=""
      iconProps={null}
      tag="button"
      text="Submit"
      type="submit"
      variant="primary"
    />
  </ReduxForm>
</Page>
`;
